cmd/ci-images-mirror/main.go

```diff
@@ -177,21 +173,7 @@ func main() {
 		logrus.WithError(err).Fatal("Failed to add imagev1 api to protobuf scheme")
 	}
 
-	mirrorStore := quayiociimagesdistributor.NewMirrorStore()
-	server := &http.Server{
-		Addr:    ":" + strconv.Itoa(opts.port),
-		Handler: getRouter(interrupts.Context(), mirrorStore),
-	}
-	interrupts.ListenAndServe(server, opts.gracePeriod)
-
-	ocClientFactory := quayiociimagesdistributor.NewClientFactory()
-	quayIOImageHelper, err := ocClientFactory.NewClient()
-	if err != nil {
-		logrus.WithError(err).Fatal("failed to create QuayIOImageHelper")
-	}
-
-	mirrorConsumerController := quayiociimagesdistributor.NewMirrorConsumer(mirrorStore, quayIOImageHelper, opts.registryConfig, opts.dryRun)
-	interrupts.Run(func(ctx context.Context) { execute(ctx, mirrorConsumerController) })
+	mirrorStore, quayIOImageHelper := mirrorstoreserver.MirrorStoreServer(opts.port, opts.gracePeriod, opts.registryConfig, false, opts.dryRun)
```

Comment:
Here I wanted to use this server also for the `pj-rehearse` so I factored it out to a separate file/package, so I could use it there, maybe it can be done better.

Reply:
Couldnt this work?

```
        ocClientFactory := quayiociimagesdistributor.NewClientFactory()
	quayIOImageHelper, err := ocClientFactory.NewClient()
	if err != nil {
		logrus.WithError(err).Fatal("failed to create QuayIOImageHelper")
	}
        // we know the requiredImageStreamTags which is basically the pairs belew
        quayIOImageHelper.ImageMirror(pairs, c.options)
```

The MirrorStore in the existing tool is there because of many tags to mirror and we want to do it as efficient as possible.
Here, for the researsal tool, there are only a couple of images to mirror.
Let us do the mirror blindly without a MirrorStore.
The code would be much simpler and we do not need the mega refactoring.